src/engraving/infrastructure/eid.h

```diff
@@ -34,33 +35,38 @@ namespace mu::engraving {
 class EID
 {
 public:
-    EID(ElementType type = ElementType::INVALID, uint32_t id = 0);
+    bool isValid() const { return *this != EID(); }
 
-    bool isValid() const { return m_type != ElementType::INVALID && m_id != 0; }
-
-    ElementType type() const { return m_type; }
-    uint32_t id() const { return m_id; }
-
-    inline bool operator ==(const EID& other) const { return m_type == other.m_type && m_id == other.m_id; }
-    inline bool operator !=(const EID& other) const { return !this->operator ==(other); }
-
-    uint64_t toUint64() const;
-    static EID fromUint64(uint64_t v);
+    inline bool operator ==(const EID& other) const { return m_first == other.m_first && m_second == other.m_second; }
+    inline bool operator !=(const EID& other) const { return m_first != other.m_first || m_second != other.m_second; }
 
     std::string toStdString() const;
     static EID fromStdString(const std::string& v);
     static EID fromStdString(const std::string_view& v);
+    static EID createNew();
 
 private:
+    uint64_t m_first = uint64_t(-1);
+    uint64_t m_second = uint64_t(-1);
 
-    ElementType m_type = ElementType::INVALID;
-    uint32_t m_id = 0;
+    friend struct std::hash<EID>;
+};
+}
+
+namespace std {
+template<>
+struct hash<mu::engraving::EID>
```

Comment:
I believe this would better be written as 
```suggestion
template<>
struct std::hash<mu::engraving::EID>
```
because otherwise some IDEs start thinking that the std namespace is ours...